File: LC/[0125] Valid Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int N = s.size();
        int ll=0, rr=N-1;

        while (ll<rr){
            while(ll<rr && !isalnum(s[ll])) ++ll;
            while(rr>ll && !isalnum(s[rr])) --rr;

            if(tolower(s[ll]) != tolower(s[rr])) return false;
            ++ll;
            --rr;
        }        
        return true;
    }
};

File: LC/[0125] Valid Palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int N=s.length();
        int ll=0, rr=N-1;

        while(ll<rr){
            while(ll<rr && !Character.isLetterOrDigit(s.charAt(ll))) ++ll;
            while(rr>ll && !Character.isLetterOrDigit(s.charAt(rr))) --rr;
            if(Character.toLowerCase(s.charAt(ll)) != Character.toLowerCase(s.charAt(rr))) return false;
            ++ll;
            --rr;
        }
        return true;
    }
}